_posts/2015-08-18-How-katas-can-help-you-learn.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+name: How-katas-can-help-you-learn
+title: How katas can help you learn
+date: 2015-08-18 09:35:00 +00:00
+author: Ana Nogal
+image:
+    src: /assets/img/custom/blog/baton_long.jpg
+    attribution:
+        text: Baton_long by Alain Delmas
+        href: https://commons.wikimedia.org/wiki/File:Baton_long.jpg
+tags:
+- Kata
+- TDD
+- iOS
+- Objective-C
+- Learning
+---
+In the last couple of weeks I have spent my time doing katas to improve my Objective-C. As you may remember, I even had an issue, [that you can revisit here](http://wp.me/p4i8Xl-7h).
+
+I've learn a lot from other katas too: for instance the RomanNumerals kata... You think it's a simple one, that you wan't learn mush from. But it's a good exercise  for your red -> green -> refactor cycle. It's a simple algorithm nothing new will come from here... Wasn't I wrong!!!!!
+
+I started  my kata, made tests then refactor. Add more tests and on the refactor phase, I decide to used an NSDictionary to map from a Decimal to a Roman number. Something like this:
+
+```
+NSDictionary *mapper = @{@10: @"X", @5:@"V", @1: @"I"};
+```
+
+That's when I discovered that the NSDictionary does not guarantee insertion order. What???? So my keys where all mixed up... And I needed them to maintain their order!!!! That's ok! I decided to create a class to map between decimals and romans. That's cool! Done! In the converter I just added a private property of type NSMutableArray. In it's constructor I added the class to the array all elements I needed. But now I have to do:
+
+```
+[[DecimalToRomanMapper alloc] initWithDecimal: AndRoman];
+```
+for every entry in the mapper. God!!!! So much work! Maybe that's why developers solve problems. They definitely don't like to do things by hand so they automate everything .
+So I decided that I could create a factory method. I really didn't know how to create this factory method, so I just looked in apple docs. Here is another thing that I've learn. Even thought I knew that it existed, I never used it because I never needed to. But as I was in learning mode, I think I was more keen to seek for a different way to do it. When you are in a client, you don't have time to experiment new things. Having this time to learn new ways of doing things is really rewarding. So here is my class:
```

Comment:
But as I was in learning mode, I think I was more keen to seek for a different way to do it.

maybe try ...

But as I was in learning mode, I think I was more interested in finding a different way to do it.